After I upgrade flutter 3.0. build error on android

/Users/gony/development/flutter/.pub-cache/hosted/pub.dartlang.org/audio_waveforms-0.1.1/android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt: (20, 1): Class 'AudioRecorder' is not abstract and does not implement abstract member public abstract fun onRequestPermissionsResult(p0: Int, p1: Array<(out) String!>, p2: IntArray): Boolean defined in io.flutter.plugin.common.PluginRegistry.RequestPermissionsResultListener
e: /Users/gony/development/flutter/.pub-cache/hosted/pub.dartlang.org/audio_waveforms-0.1.1/android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt: (97, 5): 'onRequestPermissionsResult' overrides nothing

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':audio_waveforms:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s
Exception: Gradle task assembleDebug failed with exit code 1

This is error code. How I can fix it?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,9 @@
+#[0.1.3] - 26 July, 2022
+
+- Fixed [#41](https://github.com/SimformSolutionsPvtLtd/audio_waveforms/issues/41) & [#49](https://github.com/SimformSolutionsPvtLtd/audio_waveforms/issues/49) - Upgrade Package with flutter 3.0.
+- Fixed [#50](https://github.com/SimformSolutionsPvtLtd/audio_waveforms/issues/50) & [#57](https://github.com/SimformSolutionsPvtLtd/audio_waveforms/issues/57) & [#65](https://github.com/SimformSolutionsPvtLtd/audio_waveforms/issues/65) - android build error fixed
+
+
 ## [0.1.2] - 11 May, 2022
 
 * Fixed ios resume recording issue

--- a/android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt
+++ b/android/src/main/kotlin/com/simform/audio_waveforms/AudioRecorder.kt
@@ -96,7 +96,7 @@ class AudioRecorder : PluginRegistry.RequestPermissionsResultListener {
 
     override fun onRequestPermissionsResult(
         requestCode: Int,
-        permissions: Array<out String>?,
+        permissions: Array<out String>,
         grantResults: IntArray
     ): Boolean {
         return if (requestCode == RECORD_AUDIO_REQUEST_CODE) {

--- a/example/android/app/src/main/AndroidManifest.xml
+++ b/example/android/app/src/main/AndroidManifest.xml
@@ -7,6 +7,7 @@
         <activity
             android:name=".MainActivity"
             android:launchMode="singleTop"
+            android:exported="true"
             android:theme="@style/LaunchTheme"
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
             android:hardwareAccelerated="true"

--- a/example/lib/main.dart
+++ b/example/lib/main.dart
@@ -124,7 +124,7 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
       musicFile = result.files.single.path;
       await playerController6.preparePlayer(musicFile!);
     } else {
-      print("File not picked");
+      debugPrint("File not picked");
     }
   }
 
@@ -299,7 +299,7 @@ class _HomeState extends State<Home> with WidgetsBindingObserver {
   void _playOrPlausePlayer(PlayerController controller) async {
     controller.playerState == PlayerState.playing
         ? await controller.pausePlayer()
-        : await controller.startPlayer();
+        : await controller.startPlayer(finishMode: FinishMode.loop);
   }
 
   void _startOrStopRecording() async {

--- a/example/pubspec.yaml
+++ b/example/pubspec.yaml
@@ -25,12 +25,12 @@ dependencies:
     # The example app is bundled with the plugin so we use a path dependency on
     # the parent directory to use the current plugin's version.
     path: ../
-  path_provider: ^2.0.9
-  file_picker: ^4.4.0
+  path_provider: ^2.0.11
+  file_picker: ^5.0.0
 
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
-  cupertino_icons: ^1.0.2
+  cupertino_icons: ^1.0.5
 
 dev_dependencies:
   flutter_test:
@@ -41,7 +41,7 @@ dev_dependencies:
   # activated in the `analysis_options.yaml` file located at the root of your
   # package. See that file for information about deactivating specific lint
   # rules and activating additional ones.
-  flutter_lints: ^1.0.0
+  flutter_lints: ^2.0.1
 
 # For information on the generic Dart part of this file, see the
 # following page: https://dart.dev/tools/pub/pubspec

--- a/lib/src/audio_waveforms.dart
+++ b/lib/src/audio_waveforms.dart
@@ -167,7 +167,7 @@ class _AudioWaveformsState extends State<AudioWaveforms> {
       _totalBackDistance =
           _totalBackDistance + Offset(widget.waveStyle.spacing, 0.0);
     }
-    WidgetsBinding.instance?.addPostFrameCallback((_) {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
       setState(() {});
     });
   }

--- a/lib/src/controllers/player_controller.dart
+++ b/lib/src/controllers/player_controller.dart
@@ -206,4 +206,7 @@ class PlayerController extends ChangeNotifier {
   bool operator ==(Object other) {
     return other is PlayerController && other.playerKey == playerKey;
   }
+
+  @override
+  int get hashCode => super.hashCode;
 }

--- a/lib/src/controllers/recorder_controller.dart
+++ b/lib/src/controllers/recorder_controller.dart
@@ -196,7 +196,7 @@ class RecorderController extends ChangeNotifier {
 
   ///sets [shouldClearLabels] flag to false
   void revertClearlabelCall() {
-    WidgetsBinding.instance?.addPostFrameCallback((_) {
+    WidgetsBinding.instance.addPostFrameCallback((_) {
       shouldClearLabels = false;
       notifyListeners();
     });

